src/adapters/suggest.js

```diff
@@ -1,258 +1,284 @@
-  }
-
-  favoritesRender(pois) {
-    return `
-      <h3 class="autocomplete_suggestion__category_title" onmousedown="return false;">
-        ${_('FAVORITES', 'autocomplete')}
-      </h3>
-      ${pois.map(poi => this.renderItem(poi)).join('')}`;
-  }
-
-  getValue() {
-    return this.autocomplete.getValue();
-  }
-
-  setValue(value) {
-    this.autocomplete.setValue(value);
-    this.preRender();
-  }
-
-  clear() {
-    this.autocomplete.clear();
-  }
-
-  /* select sub template */
-  renderItem(poi) {
-    const { id, name, className, subClassName, type, alternativeName } = poi;
-    const icon = IconManager.get({ className, subClassName, type });
-    const klass = `autocomplete-icon ${`icon icon-${icon.iconClass}`}`;
-    const iconDom = `<div style="color:${icon ? icon.color : ''}" class="${klass}"></div>`;
-
-    return `
-      <div class="autocomplete_suggestion"
-           data-id="${id}" data-val="${ExtendedString.htmlEncode(poi.getInputValue())}">
-        ${iconDom}
-        ${this.renderLines(name, alternativeName)}
-      </div>`;
-  }
-
-  renderCategory(category) {
-    const { label, alternativeName, color, backgroundColor } = category;
-    const icon = category.getIcon();
-    const style = `color: ${color}; background: ${backgroundColor}`;
-    const klass = `autocomplete-icon autocomplete-icon-rounded ${`icon icon-${icon.iconClass}`}`;
-    const iconDom = `<div style="${style}" class="${klass}"></div>`;
-    const categoryLabel = ExtendedString.htmlEncode(category.label);
+  renderItem(item, highlightedIndex, index) {
+    console.log('item', item);
+    if (item.id && item.id === 'geolocalisation') {
+      return item.render();
+    }
 
-    return `
-      <div class="autocomplete_suggestion autocomplete_suggestion--category"
-        data-id="${category.id}" data-val="${categoryLabel}">
-        ${iconDom}
-        ${this.renderLines(label, alternativeName)}
-      </div>`;
+    switch (item.type) {
+    case 'category':
+      return <Category category={item} selected={highlightedIndex === index} />;
+    default:
+      return <Poi poi={item} selected={highlightedIndex === index} />;
+    }
   }
 
-  renderLines(firstLabel, secondLabel) {
-    const s_firstLabel = ExtendedString.htmlEncode(firstLabel);
-    const s_secondLabel = ExtendedString.htmlEncode(secondLabel ? secondLabel : '');
-    return `
-      <div class="autocomplete_suggestion__lines_container">
-        <div class="autocomplete_suggestion__first_line">${s_firstLabel}</div>
-        <div class="autocomplete_suggestion__second_line">${s_secondLabel}</div>
-      </div>`;
+  render() {
+    const { prefixes } = this.props;
+    const { categories, pois, favorites } = this.state;
+    // console.log('pois.length', pois.length);
+
+    const items = [
+      ...prefixes,
+      ...categories,
+      ...pois.length > 0 ? pois.slice(0, SUGGEST_MAX_ITEMS - categories.length) : [],
```

Comment:
Nit: You don't need the test on length, `[ ].slice(0, x)` will always return `[ ]`.